File: 78-subsets/78-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
         List<List<Integer>> ans = new ArrayList<>();
         int n = nums.length;
         int x = 1<<n;
        int i = 0;
        while(i<x)
        {
            List<Integer> d = new ArrayList<Integer>();
            for(int j =0; j<n;j++)
            {
                if(((i>>j)&1) != 0)
                {
                    d.add(nums[j]);
                }
            }
            ans.add(d);
            i++;
        }
        return ans;
    }
}

File: 78-subsets/78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        o = []
        ans = []
        i = 0
        self.solve(nums,o,i,ans)

        
        return ans
    
    def solve(self,a, o, i, ans):
        if i == len(a):
            f = []
            for i in o:
                f.append(i)
                
            ans.append(f)
            return

            # exclude
        self.solve(a, o, i+1, ans)
            # include
        o.append(a[i])
        self.solve(a, o, i+1, ans)
        o.pop()